node/src/node_configurator/unprivileged_parse_args_configuration.rs

```diff
@@ -427,27 +427,19 @@ fn validate_mandatory_node_addr(
     }
 }
 
-fn compute_mapping_protocol_opt(
+pub fn compute_mapping_protocol_opt(
     multi_config: &MultiConfig,
     persistent_config: &mut dyn PersistentConfiguration,
     logger: &Logger,
 ) -> Option<AutomapProtocol> {
-    let persistent_mapping_protocol_opt = match persistent_config.mapping_protocol() {
-        Ok(mp_opt) => mp_opt,
-        Err(e) => {
-            warning!(
-                logger,
-                "Could not read mapping protocol from database: {:?}",
-                e
-            );
-            None
-        }
-    };
+    let persistent_mapping_protocol_opt = persistent_config
+        .mapping_protocol()
+        .expect("Error retrieving mapping protocol from CONFIG table");
     let mapping_protocol_specified = multi_config.occurrences_of("mapping-protocol") > 0;
     let computed_mapping_protocol_opt = match (
-        value_m!(multi_config, "mapping-protocol", AutomapProtocol),
-        persistent_mapping_protocol_opt,
-        mapping_protocol_specified,
+        value_m!(multi_config, "mapping-protocol", AutomapProtocol), // command line
```

Comment:
Do you insist on the little comments should stay?

Reply:
Okay, I got rid of them.